### PR TITLE
proxy: Add timeout to webauth confirmation wait

### PR DESCRIPTION
**Pull request by cloneable**
```shell
$ cargo run -p proxy --bin proxy -- --auth-backend=web --webauth-confirmation-timeout=5s
```

```
$ psql -h localhost -p 4432
NOTICE:  Welcome to Neon!
Authenticate by visiting within 5s:
    http://localhost:3000/psql_session/e946900c8a9bc6e9


psql: error: connection to server at "localhost" (::1), port 4432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 4432 failed: ERROR:  Disconnected due to inactivity after 5s.
```

### CODE REVIEW

File: proxy/src/auth/backend/web.rs [hunk 1]
```diff
@@ -41,15 +44,16 @@ impl ReportableError for WebAuthError {
     }
 }
 
-fn hello_message(redirect_uri: &reqwest::Url, session_id: &str) -> String {
+fn hello_message(redirect_uri: &reqwest::Url, session_id: &str, timeout: Duration) -> String {
     format!(
         concat![
             "Welcome to Neon!\n",
-            "Authenticate by visiting:\n",
+            "Authenticate by visiting within {timeout}:\n",
```
Comment (cloneable) [inline proxy/src/auth/backend/web.rs:51]: Do we really want to say this?

File: proxy/src/auth/backend/web.rs [hunk 2]
```diff
@@ -41,15 +44,16 @@ impl ReportableError for WebAuthError {
...
             "    {redirect_uri}{session_id}\n\n",
```
Comment (conradludgate) [inline proxy/src/auth/backend/web.rs:52]: I suppose that's not quite correct. You have to also login and choose the database within this time. Maybe something along the lines of

Authenticate by visiting
<link>
And choose a database within <time> to connect

Comment (conradludgate) [reply proxy/src/auth/backend/web.rs:52]: Yeah. Not sure about that. Perhaps instead we could make it clear at the timeout with an appropriate error message that includes the timeout duration?
"you were disconnected due to inactivity after 2 minutes" or something

Comment (stradig) [reply proxy/src/auth/backend/web.rs:52]: What is the downside of announcing? Attackers? I guess they can figure it out easily in any case...

Comment (conradludgate) [reply proxy/src/auth/backend/web.rs]: It just bloats the output

**Review by conradludgate — APPROVED** [review]

**Pull request merged**